arkit/server.py

```diff
@@ -0,0 +1,256 @@
+            return
+        threading.Timer(0.5, self.trigger_cozmo_tap).start()
+        tap_anim = random.choice(["anim_speedtap_tap_01", "anim_speedtap_tap_02", "anim_speedtap_tap_03"])
+        await self._robot.play_anim(tap_anim).wait_for_completed()
+        await self.see_fireworks(cube)
+
+    async def see_fireworks(self,cube):
+        if self._round != 2:
+            await self._robot.drive_straight(distance_mm(-50), speed_mmps(80)).wait_for_completed()
+            await self._robot.set_lift_height(0, duration=0.3).wait_for_completed()
+
+        messages = [1,2,3]
+        message = messages[self._round]
+
+        # Tap unsuccessful
+        if self._animation_triggerd is False:
+            self._trigger_message = False
+            await self._robot.play_anim("anim_keepaway_losehand_01").wait_for_completed()
+
+        # Cozmo taps cube and looks at fireworks
+        if self._animation_triggerd and self._round < 2:
+            self.send_msg(message)
+
+            # Cozmo acts excited
+            if self._round == 0:
+                await self._robot.set_head_angle(cozmo.util.Angle(degrees=44.5),1,1,1).wait_for_completed()
+                await asyncio.sleep(1)
+                await self._robot.play_anim("anim_reacttoblock_react_01_head_angle_40").wait_for_completed()
+                await self._robot.drive_straight(distance_mm(-30), speed_mmps(30)).wait_for_completed()
+                self._animation_triggerd = False
+                self._tap_action_finished = True
+
+            # Dud
+            if self._round == 1:
+                await self._robot.play_anim("anim_reacttocliff_turtlerollfail_03").wait_for_completed()
+                self._animation_triggerd = False
+                self._tap_action_finished = True
+        # Grand finale
+        if self._animation_triggerd and self._round == 2:
+            await asyncio.sleep(0.5)
+            await self._robot.play_anim("anim_memorymatch_failhand_03").wait_for_completed()
+            await asyncio.sleep(0.5)
+            await self._robot.play_anim("reacttoblock_triestoreach_01").wait_for_completed()
+            if self._animation_triggerd :
+                self.send_msg(4)
+            if self._animation_triggerd :
+                self.send_msg(3)
+            await self._robot.play_anim("anim_reacttoblock_react_01_head_angle_40").wait_for_completed()
+            await self._robot.play_anim("reacttoblock_reacttotopple_01").wait_for_completed()
+            await self._robot.play_anim("anim_reacttocliff_wheely_01").wait_for_completed()
+
+            self._tap_action_finished = True
+            await self.go_back_to_normal()
+            self._animation_triggerd = False
+            self._tap_action_finished = True
+
+    async def go_back_to_normal(self):
+        self._cube.stop_light_chaser()
+        self._cube.set_lights_off()
+        await self._robot.play_anim("anim_guarddog_getout_untouched_01").wait_for_completed()
```

Comment:
As per above, use of play_anim with specific animation name is fragile and could break with a future app version.